src/selectors.cc

```diff
@@ -420,6 +420,15 @@ select_to_reverse(const Context& context, const Selection& selection,
                   Codepoint c, int count, bool inclusive)
 {
     auto& buffer = context.buffer();
+
+    // if we are selecting backwards from the beginning of the buffer,
+    // there is nothing more that can be selected.
+    // Unless its inclusive, in which its possible that the current
+    // location of the cursor contains the Codepoint being looked for.
+    if(!inclusive && selection.cursor() == buffer.begin()){
+        return {};
+    }
```

Comment:
The inclusive motions also have a weird edge case:
If I run `kak` and type `<a-f>*`, it does not throw an error (but it should).
If I do the equivalent somewhere that's not the beginning of the buffer - for example with `iabc<esc><a-f>*` - I get "no selections remaining", which is expected.
To fix this inconsistency, I think we should apply your logic regardless of whether the motion is inclusive.

```diff
diff --git a/src/selectors.cc b/src/selectors.cc
index 3d9b453d2..60ecf07a1 100644
--- a/src/selectors.cc
+++ b/src/selectors.cc
@@ -421,4 +421,8 @@ select_to_reverse(const Context& context, const Selection& selection,
 {
     auto& buffer = context.buffer();
+
+    if (selection.cursor() == buffer.begin())
+        return {};
+
     Utf8Iterator begin{buffer.iterator_at(selection.cursor()), buffer};
     Utf8Iterator end = begin;
```

Reply:
Ah, thought I had checked that.